test/integration/connection_lost_test.exs

```diff
@@ -27,4 +27,37 @@ defmodule MongoosePush.API.ConnectionTest do
       Process.sleep(1000)
     end)
   end
+
+  test "Metrics can be correctly received from Prometheus" do
+    # Push some notification to trigger metrics
+    %{status_code: status_code, body: body} =
+      HTTPoison.post!(
+        "https://localhost:8446/v1/notification/f53453455",
+        "{\"badge\":7,\"body\":\"A message from someone\",\"click_action\":\".SomeApp.Handler.action\",\"data\":{\"custom\":\"data fields\",\"nested\":{\"fields\":\"allowed\"},\"some_id\":345645332},\"mode\":\"prod\",\"service\":\"apns\",\"tag\":\"info\",\"title\":\"Notification title\",\"topic\":\"com.someapp\"}",
```

Comment:
The payload should be a map that is encoded to JSON with a library, for better readability.